packages/hardhat-verify/src/internal/tasks/sourcify.ts

```diff
@@ -1,28 +1,179 @@
 import chalk from "chalk";
 import { subtask } from "hardhat/config";
+import { Sourcify } from "../sourcify";
+import { TASK_COMPILE } from "hardhat/builtin-tasks/task-names";
+
+import {
+  BuildInfoNotFoundError,
+  ContractNotFoundError,
+  ContractVerificationFailedError,
+  VerificationAPIUnexpectedMessageError,
+} from "../errors";
 
 import {
   TASK_VERIFY_SOURCIFY,
+  TASK_VERIFY_SOURCIFY_ATTEMPT_VERIFICATION,
   TASK_VERIFY_SOURCIFY_DISABLED_WARNING,
 } from "../task-names";
 
+interface VerificationResponse {
+  success: boolean;
+  message: string;
+}
+
+interface VerificationArgs {
+  address: string;
+  constructorArgs: string[];
+  contractFQN?: string;
+}
+
+interface VerificationInterfaceVerifyParams {
+  address: string;
+  files?: {
+    [index: string]: string;
+  };
+  chosenContract?: number;
+}
+
+interface VerificationInterface {
+  isVerified(address: string): Promise<Boolean>;
+  verify(params: VerificationInterfaceVerifyParams): Promise<any>;
+  getVerificationStatus(guid: string): Promise<any>;
+  getContractUrl(address: string, status?: string): string;
+}
+
+interface AttemptVerificationArgs {
+  address: string;
+  verificationInterface: VerificationInterface;
+  contractFQN: string;
+}
+
 /**
  * Main Sourcify verification subtask.
  *
  * Verifies a contract in Sourcify by coordinating various subtasks related
  * to contract verification.
  */
-subtask(TASK_VERIFY_SOURCIFY, async () => {
-  // code here
-});
+subtask(TASK_VERIFY_SOURCIFY)
+  .addParam("address")
+  .addFlag("listNetworks")
+  .addParam("contractFQN")
+  .setAction(
+    async (
+      { address, contractFQN }: VerificationArgs,
+      { config, network, run, artifacts }
+    ) => {
+      if (!network.config.chainId) {
```

Comment:
Sorry for leaving a comment after the initial review. I wasn't sure about this so I wanted to confirm it first. `network.config.chainId` is not reliable to get the chain id, as this value is obtained from the user config. You need to use the `eth_chainId` RPC method, like [here](https://github.com/NomicFoundation/hardhat/blob/1c472a9e6dce792f7a9a94da4db8f3db282efedf/packages/hardhat-verify/src/internal/etherscan.ts#L49).